core/src/main/java/io/grpc/Attributes.java

```diff
@@ -58,7 +60,16 @@ private Attributes() {
   @SuppressWarnings("unchecked")
   @Nullable
   public <T> T get(Key<T> key) {
-    return (T) data.get(key.name);
+    return (T) data.get(key);
+  }
+
+  /**
+   * Returns size of the container.
+   *
+   * @return int size of the container.
+   */
+  public int size() {
```

Comment:
This doesn't make sense in the public API. There are no other means by which to iterate through the attributes, so this isn't useful.

Maybe make it package-private and mark it `@VisibleForTesting`?